gcc/d/gdc.texi

```diff
@@ -120,196 +120,662 @@ and the Info entries for @file{gdc} and @file{gcc}.
+when a function in the @code{core.stdc} package is a built-in function.
 
-@item -fd-vtls
-@cindex @option{-fd-vtls}
-List all variables going into thread local storage.
+@item -fdebug
+@item -fdebug=@var{value}
+@cindex @option{-fdebug}
+@cindex @option{-fno-debug}
+Turn on compilation of conditional @code{debug} code into the program.
+The @option{-fdebug} option itself sets the debug level to @code{1},
+while @option{-fdebug=} enables @code{debug} code that are identified
+by any of the following values:
 
-@item -fignore-unknown-pragmas
-@cindex @option{-fignore-unknown-pragmas}
-Ignore unsupported pragmas.
+@table @samp
+@item level
+@cindex @option{-fdebug=level}
+Sets the debug level to @var{level}, any @code{debug} code <= @var{level}
+is compiled into the program.
+@item ident
+@cindex @option{-fdebug=ident}
+Turns on compilation of any @code{debug} code identified by @var{ident}.
+@end table
 
-@item -fsplit-dynamic-arrays
-@cindex @option{-fsplit-dynamic-arrays}
-Split dynamic arrays into length and pointer when passing to functions.
+@item -fno-invariants
+@cindex @option{-finvariants}
+@cindex @option{-fno-invariants}
+Turns off code generation for class @code{invariant} contracts.
+
+@item -fno-moduleinfo
+@cindex @option{-fmoduleinfo}
+@cindex @option{-fno-moduleinfo}
+Turns off generation of the @code{ModuleInfo} and related functions
+that would become unreferenced without it, which may allow linking
+to programs not written in D.  Functions that will not be generated
+include module constructor and destructors (@code{static this} and
+@code{static ~this}), @code{unittest} code, and @code{DSO} registry
+functions for dynamically linked code.
 
-@item -femit-templates
-@cindex @option{-femit-templates}
-Control template emission behaviour.
+@item -fonly=@var{filename}
+@cindex @option{-fonly}
+Tells the compiler to parse and run semantic analysis on all modules
+on the command line, but only generate code for the module specified
+by @var{filename}.
 
-Turning on this option will emit all templates, but make them private
-to the translation unit. The executable will have multiple copies
-of code and data.
+@item -fno-postconditions
+@cindex @option{-fpostconditions}
+@cindex @option{-fno-postconditions}
+Turns off code generation for postcondition @code{out} contracts.
 
-Turning off this option will tell the compiler to not emit templates
-at all.
+@item -fno-preconditions
+@cindex @option{-fpreconditions}
+@cindex @option{-fno-preconditions}
+Turns off code generation for precondition @code{in} contracts.
 
-The default behaviour is to emit templates, but only if the compiler
-determines that it needs to.
+@item -frelease
+@cindex @option{-frelease}
+@cindex @option{-fno-release}
+Turns on compiling in release mode, which means not emitting runtime
+checks for contracts and asserts.  Array bounds checking is not done
+for @code{@@system} and @code{@@trusted} functions, and assertion
+failures are undefined behaviour.
```

Comment:
`undefined behaviour` we do not make use of this right now, correct? We could probably add a `if(...) __builtin_unreachable` at some time but I wonder how much code this will break ;-)

It's probably nevertheless a good idea to document this as `undefined behaviour`.